### DISCUSSION TITLE
Set workflow "Cancelled" instead of "Finished"

### DISCUSSION THREAD

**Discussion by aaronfhamilton** [elsa-workflows/elsa-core#3410] [Q&A]
How do I mark short running workflow as "Cancelled" instead of "Finished"? I'm using a RabbitMQ trigger to trigger workflows for messages having only certain values, where 90% of the messages are not relevant. Since I can't filter the messages before starting the workflow, is it possible to set the status to cancelled instead of finished (or something other than finished)?

Ideally, being able to also immediately delete these instances would be also helpful.

Thanks!

**Comment by sfmskywalker** [comment]
We could add a Cancel activity that cancels the workflow. You could probably implement a custom activity that does this right now. You might take a look at the source code to see how workflows are cancelled via the API. 

Automatically deleting a workflow instance could be done using a retention policy that executes every minute for example.

Alternatively, you could implement a custom notification handler that handles a WorkflowCompleted and/or WorkflowCancelled event and delete the workflow instance from there immediately.

Elsa 3 (under development) will not create workflow instances unless you want to, so there you would only need an API or activity to cancel a running workflow.
